subsys/bluetooth/shell/bt.c

```diff
@@ -1245,6 +1243,26 @@ static void bt_ready(int err)
 #endif /* CONFIG_BT_SMP */
 }
 
+static void active_scan_timeout(struct k_work *work)
+{
+	int err;
+
+	shell_print(ctx_shell, "Scan timeout");
+
+	err = bt_le_scan_stop();
+	if (err) {
+		shell_error(ctx_shell, "Failed to stop scan (err %d)", err);
+	}
+
+#if defined(CONFIG_BT_CENTRAL)
+	if (auto_connect.connect_name) {
+		auto_connect.connect_name = false;
+		/* "name" is what would be in argv[0] normally */
+		cmd_scan_filter_clear_name(ctx_shell, 1, (char *[]){ "name" });
+	}
+#endif /* CONFIG_BT_CENTRAL */
+}
```

Comment:
I think that if you move this file to before line 228 then you can omit the prototype of it, right?

Reply:
Yes, do you think it would be better if I moved it?

Reply:
Slightly, but not necessary :)

Reply:
Ok :)